Long overflow in UPDATE should error gracefully

```
➜  ~ psql -h localhost -p 5432
psql (16.3, server 16)
Type "help" for help.

jdt=> INSERT into foo (_id, val) VALUES (1, 1);
INSERT 0 0
jdt=> UPDATE foo SET val = (9223372036854775807 + 9223372036854775807) WHERE _id = 1;
UPDATE 0
jdt=> SELECT * FROM foo;
 _id | val 
-----+-----
   1 |   1
(1 row)

jdt=> SELECT * FROM _txs;
 _id | committed | error |             tx_time              
-----+-----------+-------+----------------------------------
   0 | t         |       | 2024-08-06 11:09:29.822173+00:00
(1 row)

jdt=> 
```

```
11:09:34 | ERROR xtdb.indexer | error in indexer
java.lang.ArithmeticException: long overflow
        at java.base/java.lang.Math.addExact(Math.java:931)
        at clojure.core$eval34257$fn__34258.invoke(NO_SOURCE_FILE:0)
        at xtdb.expression$$reify__13915.project(expression.clj:1555)
        at xtdb.operator.project.ProjectCursor$reify__20819.accept(project.clj:124)
        at xtdb.operator.rename.RenameCursor$reify__21628.accept(rename.clj:39)
        at xtdb.operator.scan.TrieCursor$fn__19489.invoke(scan.clj:309)
        at xtdb.operator.scan.TrieCursor.tryAdvance(scan.clj:260)
        at xtdb.operator.rename.RenameCursor.tryAdvance(rename.clj:29)
        at xtdb.operator.project.ProjectCursor.tryAdvance(project.clj:116)
        at xtdb.operator.project.ProjectCursor.tryAdvance(project.clj:116)
        at xtdb.query$wrap_cursor$reify__25716.tryAdvance(query.clj:78)
        at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:332)
        at xtdb.indexer$query_indexer$eval_query__25954.invoke(indexer.clj:396)
        at xtdb.indexer$wrap_sql_args$fn__25961.invoke(indexer.clj:430)
        at xtdb.indexer$foreach_arg_row.invokeStatic(indexer.clj:413)
        at xtdb.indexer$foreach_arg_row.invokePrim(indexer.clj)
        at xtdb.indexer$$reify__25966.indexOp(indexer.clj:401)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019$fn__26032.invoke(indexer.clj:623)
        at clojure.lang.AFn.call(AFn.java:18)
        at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019.invoke(indexer.clj:620)
        at xtdb.indexer.Indexer$fn__26016$fn__26040.invoke(indexer.clj:634)
        at xtdb.indexer.Indexer$fn__26016.invoke(indexer.clj:633)
        at xtdb.indexer.Indexer.indexTx(indexer.clj:597)
        at xtdb.log.watcher$watch_log_BANG_$reify__26256.acceptRecord(watcher.clj:53)
        at xtdb.log$tx_handler$fn__17626.invoke(log.clj:37)
        at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
        at clojure.core.protocols$fn__8230.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8230.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at xtdb.log.NotifyingSubscriberHandler$fn__17659.invoke(log.clj:96)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.lang.Thread.run(Thread.java:1583)
11:09:34 | ERROR xtdb.util | Uncaught exception:
java.lang.ArithmeticException: long overflow
        at java.base/java.lang.Math.addExact(Math.java:931)
        at clojure.core$eval34257$fn__34258.invoke(NO_SOURCE_FILE:0)
        at xtdb.expression$$reify__13915.project(expression.clj:1555)
        at xtdb.operator.project.ProjectCursor$reify__20819.accept(project.clj:124)
        at xtdb.operator.rename.RenameCursor$reify__21628.accept(rename.clj:39)
        at xtdb.operator.scan.TrieCursor$fn__19489.invoke(scan.clj:309)
        at xtdb.operator.scan.TrieCursor.tryAdvance(scan.clj:260)
        at xtdb.operator.rename.RenameCursor.tryAdvance(rename.clj:29)
        at xtdb.operator.project.ProjectCursor.tryAdvance(project.clj:116)
        at xtdb.operator.project.ProjectCursor.tryAdvance(project.clj:116)
        at xtdb.query$wrap_cursor$reify__25716.tryAdvance(query.clj:78)
        at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:332)
        at xtdb.indexer$query_indexer$eval_query__25954.invoke(indexer.clj:396)
        at xtdb.indexer$wrap_sql_args$fn__25961.invoke(indexer.clj:430)
        at xtdb.indexer$foreach_arg_row.invokeStatic(indexer.clj:413)
        at xtdb.indexer$foreach_arg_row.invokePrim(indexer.clj)
        at xtdb.indexer$$reify__25966.indexOp(indexer.clj:401)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019$fn__26032.invoke(indexer.clj:623)
        at clojure.lang.AFn.call(AFn.java:18)
        at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019.invoke(indexer.clj:620)
        at xtdb.indexer.Indexer$fn__26016$fn__26040.invoke(indexer.clj:634)
        at xtdb.indexer.Indexer$fn__26016.invoke(indexer.clj:633)
        at xtdb.indexer.Indexer.indexTx(indexer.clj:597)
        at xtdb.log.watcher$watch_log_BANG_$reify__26256.acceptRecord(watcher.clj:53)
        at xtdb.log$tx_handler$fn__17626.invoke(log.clj:37)
        at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
        at clojure.core.protocols$fn__8230.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8230.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at xtdb.log.NotifyingSubscriberHandler$fn__17659.invoke(log.clj:96)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.lang.Thread.run(Thread.java:1583)
11:09:34 | DEBUG xtdb.pgwire | Error on execute-tx
java.util.concurrent.ExecutionException: clojure.lang.ExceptionInfo: Ingestion stopped: long overflow {}
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
        at clojure.core$deref_future.invokeStatic(core.clj:2317)
        at clojure.core$deref.invokeStatic(core.clj:2337)
        at clojure.core$deref.invoke(core.clj:2323)
        at xtdb.indexer.Indexer.awaitTx(indexer.clj:568)
        at xtdb.node.impl.Node.prepareQuery(impl.clj:133)
        at xtdb.node.impl.Node.open_sql_query(impl.clj:115)
        at xtdb.node.impl.Node.execute_tx(impl.clj:105)
        at xtdb.api$execute_tx.invokeStatic(api.clj:208)
        at xtdb.api$execute_tx.invoke(api.clj:180)
        at xtdb.pgwire$execute_tx.invokeStatic(pgwire.clj:1077)
        at xtdb.pgwire$execute_tx.invoke(pgwire.clj:1072)
        at xtdb.pgwire$cmd_exec_dml.invokeStatic(pgwire.clj:1109)
        at xtdb.pgwire$cmd_exec_dml.invoke(pgwire.clj:1095)
        at xtdb.pgwire$cmd_execute.invokeStatic(pgwire.clj:1437)
        at xtdb.pgwire$cmd_execute.invoke(pgwire.clj:1418)
        at xtdb.pgwire$cmd_simple_query.invokeStatic(pgwire.clj:1467)
        at xtdb.pgwire$cmd_simple_query.invoke(pgwire.clj:1442)
        at xtdb.pgwire$handle_msg.invokeStatic(pgwire.clj:1494)
        at xtdb.pgwire$handle_msg.invoke(pgwire.clj:1475)
        at xtdb.pgwire$conn_loop.invokeStatic(pgwire.clj:1561)
        at xtdb.pgwire$conn_loop.invoke(pgwire.clj:1514)
        at xtdb.pgwire$connect.invokeStatic(pgwire.clj:1597)
        at xtdb.pgwire$connect.invoke(pgwire.clj:1564)
        at xtdb.pgwire$accept_loop$fn__34137$fn__34138$fn__34139.invoke(pgwire.clj:1652)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: clojure.lang.ExceptionInfo: Ingestion stopped: long overflow
        at xtdb.await$__GT_ingester_ex.invokeStatic(await.clj:18)
        at xtdb.await$__GT_ingester_ex.invoke(await.clj:17)
        at xtdb.await$notify_ex.invokeStatic(await.clj:60)
        at xtdb.await$notify_ex.invoke(await.clj:56)
        at xtdb.indexer.Indexer.indexTx(indexer.clj:665)
        at xtdb.log.watcher$watch_log_BANG_$reify__26256.acceptRecord(watcher.clj:53)
        at xtdb.log$tx_handler$fn__17626.invoke(log.clj:37)
        at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
        at clojure.core.protocols$fn__8230.invokeStatic(protocols.clj:75)
        at clojure.core.protocols$fn__8230.invoke(protocols.clj:75)
        at clojure.core.protocols$fn__8178$G__8173__8191.invoke(protocols.clj:13)
        at clojure.core$reduce.invokeStatic(core.clj:6886)
        at clojure.core$reduce.invoke(core.clj:6868)
        at xtdb.log.NotifyingSubscriberHandler$fn__17659.invoke(log.clj:96)
        at clojure.lang.AFn.run(AFn.java:22)
        ... 1 common frames omitted
Caused by: java.lang.ArithmeticException: long overflow
        at java.base/java.lang.Math.addExact(Math.java:931)
        at clojure.core$eval34257$fn__34258.invoke(NO_SOURCE_FILE:0)
        at xtdb.expression$$reify__13915.project(expression.clj:1555)
        at xtdb.operator.project.ProjectCursor$reify__20819.accept(project.clj:124)
        at xtdb.operator.rename.RenameCursor$reify__21628.accept(rename.clj:39)
        at xtdb.operator.scan.TrieCursor$fn__19489.invoke(scan.clj:309)
        at xtdb.operator.scan.TrieCursor.tryAdvance(scan.clj:260)
        at xtdb.operator.rename.RenameCursor.tryAdvance(rename.clj:29)
        at xtdb.operator.project.ProjectCursor.tryAdvance(project.clj:116)
        at xtdb.operator.project.ProjectCursor.tryAdvance(project.clj:116)
        at xtdb.query$wrap_cursor$reify__25716.tryAdvance(query.clj:78)
        at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:332)
        at xtdb.indexer$query_indexer$eval_query__25954.invoke(indexer.clj:396)
        at xtdb.indexer$wrap_sql_args$fn__25961.invoke(indexer.clj:430)
        at xtdb.indexer$foreach_arg_row.invokeStatic(indexer.clj:413)
        at xtdb.indexer$foreach_arg_row.invokePrim(indexer.clj)
        at xtdb.indexer$$reify__25966.indexOp(indexer.clj:401)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019$fn__26032.invoke(indexer.clj:623)
        at clojure.lang.AFn.call(AFn.java:18)
        at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175)
        at xtdb.indexer.Indexer$fn__26016$index_tx_ops__26019.invoke(indexer.clj:620)
        at xtdb.indexer.Indexer$fn__26016$fn__26040.invoke(indexer.clj:634)
        at xtdb.indexer.Indexer$fn__26016.invoke(indexer.clj:633)
        at xtdb.indexer.Indexer.indexTx(indexer.clj:597)
        ... 11 common frames omitted
```

--- a/core/src/main/clojure/xtdb/expression.clj
+++ b/core/src/main/clojure/xtdb/expression.clj
@@ -818,12 +818,7 @@
 
 (defmethod codegen-call [:/ :int :int] [{:keys [arg-types]}]
   {:return-type (types/least-upper-bound arg-types)
-   :->call-code #(do `(try
-                        (quot ~@%)
-                        (catch ArithmeticException e#
-                          (case (.getMessage e#)
-                            "/ by zero" (throw-div-0)
-                            (throw e#)))))})
+   :->call-code #(do `(quot ~@%))})
 
 (defmethod codegen-call [:/ :num :num] [{:keys [arg-types]}]
   {:return-type (types/least-upper-bound arg-types)
@@ -1615,6 +1610,19 @@
   (fn [expr opts]
     (f expr (assoc opts :zone-id (.getZone *clock*)))))
 
+(defn arithmetic-ex->runtime-ex [^Throwable cause]
+  (let [message (.getMessage cause)]
+    (cond
+      (and message (str/index-of message "/ by zero"))
+      (err/runtime-err ::division-by-zero {::err/message "data exception - division by zero"} cause)
+
+      (and message (str/index-of message "overflow"))
+      (err/runtime-err ::overflow-error {::err/message "data exception - overflow error"} cause)
+
+      :else
+      (err/runtime-err ::unknown-arithmetic-error {::err/message "data exception - arithmetic exception"} cause))))
+
+
 (def ^:private emit-projection
   "NOTE: we macroexpand inside the memoize on the assumption that
    everything outside yields the same result on the pre-expanded expr - this
@@ -1636,7 +1644,6 @@
                                       (dotimes [~idx-sym row-count#]
                                         ~(continue (fn [t c]
                                                      (write-value-out! t c))))))
-
                                  #_(doto clojure.pprint/pprint) ; <<no-commit>>
                                  #_(->> (binding [*print-meta* true]))
                                  eval))
@@ -1679,7 +1686,10 @@
             (doto out-vec
               (.setInitialCapacity row-count)
               (.allocateNew))
-            (@!projection-fn in-rel schema params out-vec)
+            (try
+              (@!projection-fn in-rel schema params out-vec)
+              (catch ArithmeticException e
+                (throw (arithmetic-ex->runtime-ex e))))
             (.setValueCount out-vec row-count)
             (vr/vec->reader out-vec)))))))
 

--- a/core/src/main/kotlin/xtdb/vector/FieldVectorWriters.kt
+++ b/core/src/main/kotlin/xtdb/vector/FieldVectorWriters.kt
@@ -215,10 +215,19 @@ private class TimeNanoVectorWriter(override val vector: TimeNanoVector) : TimeVe
     override fun writeLong(v: Long) = vector.setSafe(wp.getPositionAndIncrement(), v)
 }
 
+private val DECIMAL_ERROR_KEY = Keyword.intern("xtdb.error", "decimal-error")
+
 private class DecimalVectorWriter(override val vector: DecimalVector) : ScalarVectorWriter(vector) {
+    @Throws(RuntimeException::class)
     override fun writeObject0(obj: Any) {
         if (obj !is BigDecimal) throw InvalidWriteObjectException(field, obj)
-        vector.setSafe(wp.getPositionAndIncrement(), obj.setScale(vector.scale))
+        try {
+            vector.setSafe(wp.getPositionAndIncrement(), obj.setScale(vector.scale))
+        } catch (e: Exception) {
+            throw RuntimeException(DECIMAL_ERROR_KEY, e.message, emptyMap<Keyword, Any>(), e)
+        } catch (e: ArithmeticException) {
+            throw RuntimeException(DECIMAL_ERROR_KEY, e.message, emptyMap<Keyword, Any>(), e)
+        }
     }
 
     override fun writeValue0(v: ValueReader) = writeObject(v.readObject())

--- a/http-server/src/test/clojure/xtdb/remote_test.clj
+++ b/http-server/src/test/clojure/xtdb/remote_test.clj
@@ -128,7 +128,7 @@
           body (decode-json body)]
       (t/is (= 200 status))
       (t/is (instance? xtdb.RuntimeException body))
-      (t/is (= "data exception — division by zero" (ex-message body)))
+      (t/is (= "data exception - division by zero" (ex-message body)))
       (t/is (= {:xtdb.error/error-key :xtdb.expression/division-by-zero}
                (ex-data body)))))
 

--- a/src/test/clojure/xtdb/api_test.clj
+++ b/src/test/clojure/xtdb/api_test.clj
@@ -551,7 +551,7 @@ VALUES (2, DATE '2022-01-01', DATE '2021-01-01')"]])
                           (xt/q tu/*node* '(from docs [name]))))
 
   (t/is (thrown-with-msg? xtdb.RuntimeException
-                          #"data exception — division by zero"
+                          #"data exception - division by zero"
                           (xt/q tu/*node* '(-> (rel [{}] [])
                                                (with {:foo (/ 1 0)})))))
 

--- a/src/test/clojure/xtdb/expression_test.clj
+++ b/src/test/clojure/xtdb/expression_test.clj
@@ -1392,15 +1392,17 @@
     (t/is (= {:res 3.0, :res-type [:union #{:f64 :decimal}]}
              (run-test 'greatest (double 3.0) (bigdec 2.0))))
 
+
     ;; TODO decide on behaviour
     ;; out of fixed precision range
-    (t/is (thrown? UnsupportedOperationException
+    (t/is (thrown? RuntimeException
                    (run-test '+ (bigdec 1E+19M) (bigdec 1E+19M))))
     ;; overflow
-    (t/is (thrown? UnsupportedOperationException
+    (t/is (thrown? RuntimeException
                    (run-test '+ (bigdec 1E+35M) (bigdec 1E-35M))))
+
     ;; underflow
-    (t/is (thrown? ArithmeticException
+    (t/is (thrown? RuntimeException
                    (run-test '/ (bigdec 1E-35M) (bigdec 1E+35M))))))
 
 (t/deftest test-throws-on-overflow

--- a/src/test/clojure/xtdb/expression_test.clj
+++ b/src/test/clojure/xtdb/expression_test.clj
@@ -387,7 +387,11 @@
   (t/is (thrown-with-msg? RuntimeException #"division by zero"
                           (project1 '(/ a 0.0) {:a 5})))
   (t/is (thrown-with-msg? RuntimeException #"division by zero"
-                          (project1 '(/ a 0) {:a 5.0}))))
+                          (project1 '(/ a 0) {:a 5.0})))
+  (t/is (thrown-with-msg? RuntimeException #"overflow"
+                          (project1 '(+ a 9223372036854775807) {:a 9223372036854775807})))
+  (t/is (thrown-with-msg? RuntimeException #"overflow"
+                          (project1 '(- a -9223372036854775807) {:a 9223372036854775807}))))
 
 (defn- project-mono-value [f-sym val col-type]
   (with-open [rel (tu/open-rel [(tu/open-vec (types/col-type->field "s"  col-type) [val])])]
@@ -1411,16 +1415,16 @@
               (-> (run-projection rel (list f 'x 'y))
                   (update :res first))))]
 
-    (t/is (thrown? ArithmeticException
+    (t/is (thrown? RuntimeException
                    (run-binary-test '+ (Integer/MAX_VALUE) (int 4))))
 
-    (t/is (thrown? ArithmeticException
+    (t/is (thrown? RuntimeException
                    (run-binary-test '- (Integer/MIN_VALUE) (int 4))))
 
-    (t/is (thrown? ArithmeticException
+    (t/is (thrown? RuntimeException
                    (run-unary-test '- (Integer/MIN_VALUE))))
 
-    (t/is (thrown? ArithmeticException
+    (t/is (thrown? RuntimeException
                    (run-binary-test '* (Integer/MIN_VALUE) (int 2))))
 
     #_ ; TODO this one throws IAE because that's what clojure.lang.Numbers/shortCast throws
@@ -1507,10 +1511,10 @@
                                 #(->dur-vec "x" TimeUnit/SECOND 154)
                                 #(->ts-vec "y" TimeUnit/NANOSECOND 1234))))
 
-      (t/is (thrown? ArithmeticException
-                     (test-projection '+
-                                      #(->ts-vec "x" TimeUnit/MILLISECOND (- Long/MAX_VALUE 500))
-                                      #(->dur-vec "y" TimeUnit/SECOND 1))))
+      (t/is (thrown-with-msg? RuntimeException  #"data exception - overflow error"
+                              (test-projection '+
+                                               #(->ts-vec "x" TimeUnit/MILLISECOND (- Long/MAX_VALUE 500))
+                                               #(->dur-vec "y" TimeUnit/SECOND 1))))
 
       (t/is (= {:res [(time/->zdt #inst "2020-12-31T23:59:59.998Z")]
                 :res-type [:timestamp-tz :micro "UTC"]}

--- a/src/test/clojure/xtdb/pgwire_test.clj
+++ b/src/test/clojure/xtdb/pgwire_test.clj
@@ -157,11 +157,11 @@
 
 ;;TODO ADD support for multiple statments in a single simple query
 #_(deftest mulitiple-statement-simple-query-test
-  (with-open [conn (jdbc-conn "preferQueryMode" "simple")
-              stmt (.createStatement conn)
-              rs (.executeQuery stmt "SELECT a.a FROM (VALUES ('hello, world')) a (a)")]
-    (is (= true (.next rs)))
-    (is (= false (.next rs)))))
+    (with-open [conn (jdbc-conn "preferQueryMode" "simple")
+                stmt (.createStatement conn)
+                rs (.executeQuery stmt "SELECT a.a FROM (VALUES ('hello, world')) a (a)")]
+      (is (= true (.next rs)))
+      (is (= false (.next rs)))))
 
 (deftest prepared-query-test
   (with-open [conn (jdbc-conn "prepareThreshold" "1")
@@ -618,7 +618,7 @@
        (testing "error during query execution"
          (with-redefs [clojure.tools.logging/logf (constantly nil)]
            (send "select (1 / 0) from (values (42)) a (a);\n")
-           (is (= [["ERROR:  data exception — division by zero"]] (read :err))))
+           (is (= [["ERROR:  data exception - division by zero"]] (read :err))))
 
          (testing "query error allows session to continue"
            (send "select 'ping';\n")
@@ -798,17 +798,17 @@
   (with-open [conn (jdbc-conn)]
     (q conn ["BEGIN"])
     (is (thrown-with-msg?
-          PSQLException
-          #"ERROR\: invalid transaction state \-\- active SQL\-transaction"
-          (q conn ["SET TRANSACTION READ WRITE"])))))
+         PSQLException
+         #"ERROR\: invalid transaction state \-\- active SQL\-transaction"
+         (q conn ["SET TRANSACTION READ WRITE"])))))
 
 (deftest begin-in-a-transaction-error-test
   (with-open [conn (jdbc-conn)]
     (q conn ["BEGIN"])
     (is (thrown-with-msg?
-          PSQLException
-          #"ERROR\: invalid transaction state \-\- active SQL\-transaction"
-          (q conn ["BEGIN"])))))
+         PSQLException
+         #"ERROR\: invalid transaction state \-\- active SQL\-transaction"
+         (q conn ["BEGIN"])))))
 
 (deftest test-current-time
   ;; no support for setting current-time so need to interact with clock directly
@@ -992,8 +992,8 @@
       (sql "COMMIT")
 
       (is (= [{:version 1,
-              :_valid_from #inst "2020-01-02T00:00:00.000000000-00:00",
-              :_valid_to nil}]
+               :_valid_from #inst "2020-01-02T00:00:00.000000000-00:00",
+               :_valid_to nil}]
              (q conn ["SELECT version, _valid_from, _valid_to FROM foo"])))
 
       (is (= [{:version 0,
@@ -1347,7 +1347,7 @@
 
       (.setObject stmt 1 nil)
 
-        (t/is (= ["text"] (param-metadata stmt)))
+      (t/is (= ["text"] (param-metadata stmt)))
 
       (with-open [rs (.executeQuery stmt)]
 
@@ -1559,18 +1559,18 @@
 
   (when (psql-available?)
     (psql-session
-       (fn [send read]
-         (send "INSERT INTO bar RECORDS {_id: 1, v: 1};\n")
-         (read)
-         (send "INSERT INTO bar RECORDS {_id: 1, v: 2};\n")
-         (read)
+     (fn [send read]
+       (send "INSERT INTO bar RECORDS {_id: 1, v: 1};\n")
+       (read)
+       (send "INSERT INTO bar RECORDS {_id: 1, v: 2};\n")
+       (read)
 
-         (send "SELECT *, _valid_time FROM bar FOR ALL VALID_TIME;\n")
+       (send "SELECT *, _valid_time FROM bar FOR ALL VALID_TIME;\n")
 
-         (t/is (= [["_id" "v" "_valid_time"]
-                   ["1" "2" "[2020-01-04 00:00:00+00:00,)"]
-                   ["1" "1" "[2020-01-03 00:00:00+00:00,2020-01-04 00:00:00+00:00)"]]
-                  (read)))))))
+       (t/is (= [["_id" "v" "_valid_time"]
+                 ["1" "2" "[2020-01-04 00:00:00+00:00,)"]
+                 ["1" "1" "[2020-01-03 00:00:00+00:00,2020-01-04 00:00:00+00:00)"]]
+                (read)))))))
 
 (when (psql-available?)
   (deftest test-datetime-formatting
@@ -1582,49 +1582,49 @@
       (psql-session
        (fn [send read]
 
-           (t/testing "timestamps are correctly output in text format"
-             ;;note nanosecond timestamp is returned as json
-             (send "SET TIME ZONE '+03:21';\n")
-             (read)
-             (send "SHOW timezone;\n")
-             (t/is (= [["timezone"] ["+03:21"]] (read)))
+         (t/testing "timestamps are correctly output in text format"
+           ;;note nanosecond timestamp is returned as json
+           (send "SET TIME ZONE '+03:21';\n")
+           (read)
+           (send "SHOW timezone;\n")
+           (t/is (= [["timezone"] ["+03:21"]] (read)))
 
-             (send "SELECT
+           (send "SELECT
                     TIMESTAMP '3000-04-15T20:40:31+01:00[Europe/London]' zdt,
                     CURRENT_DATE cd,
                     CURRENT_TIMESTAMP cts, CURRENT_TIMESTAMP(4) cts4,
                     LOCALTIMESTAMP lts, LOCALTIMESTAMP(9) lts9;\n")
 
-             (t/is (=
-                    [["zdt" "cd" "cts" "cts4" "lts" "lts9"]
-                     ["3000-04-15 20:40:31+01:00"
-                      "2022-08-16"
-                      "2022-08-16 14:29:03.123456+03:21"
-                      "2022-08-16 14:29:03.1234+03:21"
-                      "2022-08-16 14:29:03.123456"
-                      "\"2022-08-16T14:29:03.123456789\""]]
-                    (read)))
+           (t/is (=
+                  [["zdt" "cd" "cts" "cts4" "lts" "lts9"]
+                   ["3000-04-15 20:40:31+01:00"
+                    "2022-08-16"
+                    "2022-08-16 14:29:03.123456+03:21"
+                    "2022-08-16 14:29:03.1234+03:21"
+                    "2022-08-16 14:29:03.123456"
+                    "\"2022-08-16T14:29:03.123456789\""]]
+                  (read)))
 
-             (send "SET TIME ZONE 'GMT';\n")
-             (read)
-             (send "SHOW timezone;\n")
-             (t/is (= [["timezone"] ["GMT"]] (read))))
+           (send "SET TIME ZONE 'GMT';\n")
+           (read)
+           (send "SHOW timezone;\n")
+           (t/is (= [["timezone"] ["GMT"]] (read))))
 
-             (send "SELECT
+         (send "SELECT
                     TIMESTAMP '3000-04-15T20:40:31+01:00[Europe/London]' zdt,
                     CURRENT_DATE cd,
                     CURRENT_TIMESTAMP cts, CURRENT_TIMESTAMP(4) cts4,
                     LOCALTIMESTAMP lts, LOCALTIMESTAMP(9) lts9;\n")
 
-             (t/is (=
-                    [["zdt" "cd" "cts" "cts4" "lts" "lts9"]
-                     ["3000-04-15 20:40:31+01:00"
-                      "2022-08-16"
-                      "2022-08-16 11:08:03.123456+00:00"
-                      "2022-08-16 11:08:03.1234+00:00"
-                      "2022-08-16 11:08:03.123456"
-                      "\"2022-08-16T11:08:03.123456789\""]]
-                    (read))))))))
+         (t/is (=
+                [["zdt" "cd" "cts" "cts4" "lts" "lts9"]
+                 ["3000-04-15 20:40:31+01:00"
+                  "2022-08-16"
+                  "2022-08-16 11:08:03.123456+00:00"
+                  "2022-08-16 11:08:03.1234+00:00"
+                  "2022-08-16 11:08:03.123456"
+                  "\"2022-08-16T11:08:03.123456789\""]]
+                (read))))))))
 
 (deftest test-pg
   (with-open [conn (pg-conn {})]
@@ -1669,7 +1669,7 @@
              PGErrorResponse
              #"Missing types for params - client must specify types for all params in DML statements"
              (pg/execute conn "INSERT INTO foo(_id, v) VALUES (1, $1)" {:params ["1"]
-                                                                          :oids [OID/DEFAULT]}))
+                                                                        :oids [OID/DEFAULT]}))
             "params declared with the default oid (0) by clients are
              treated as unspecified and therefore also error"))))
 
@@ -1855,7 +1855,7 @@ FROM pg_catalog.pg_type t
   JOIN pg_catalog.pg_namespace n ON t.typnamespace = n.oid
 WHERE t.typname = $1 AND (n.nspname = $2 OR $3 AND n.nspname = ANY (current_schemas(true)))
 ORDER BY t.oid DESC LIMIT 1"
-                           {:args ["transit" nil true]})))))
+                     {:args ["transit" nil true]})))))
 
   (with-open [conn (jdbc-conn "prepareThreshold" -1)]
     (jdbc/execute! conn ["INSERT INTO foo (_id, v) VALUES (1, ?)" (pgwire/transit->pgobject {:a 1, :b 2})])

